dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilderUtility.h

```diff
@@ -96,26 +78,41 @@ inline PCWSTR GetWinSoundEventString(AppNotificationBuilder::AppNotificationSoun
     }
 }
 
-inline std::wstring EncodeArgument(std::wstring const& value)
+inline std::wstring GetArgumentEncoding(wchar_t const& ch)
 {
-    std::wstring encodedValue{};
+    static std::unordered_map<wchar_t, std::wstring> encodings = { { L'&', L"&amp;"}, { L'\"', L"&quot;"}, {L'<', L"&lt;"},
+                                                                        {L'>', L"&gt;"}, {L'\'', L"&apos;"}, { L'%', L"%25"},
+                                                                            {L';', L"%3B"}, {L'=', L"%3D"} };
+    if (encodings.find(ch) != encodings.end())
+    {
+        return encodings[ch];
+    }
+    else
+    {
+        return std::wstring(1, ch);
+    }
+}
 
-    auto percentEncodings{ GetPercentEncodings() };
-    auto xmlEncodings{ GetXmlEscapeEncodings() };
+inline std::wstring GetXmlEncoding(wchar_t const& ch)
+{
+    static std::unordered_map<wchar_t, std::wstring> encodings = { { L'&', L"&amp;"}, { L'\"', L"&quot;"}, {L'<', L"&lt;"},
```

Comment:
I notice we have a mismatch in number of mappings for both encoding and decoding map. Is that on purpose? It should be the same right?

Reply:
Or are we saying Xml and Argument encoding for win32 are different?

Reply:
Argument encoding does two things:
- Encodes the essential XML characters (& " ' < >)
- Encodes the argument to the format we want (% = ;)

The platform already decodes the XML characters for us. The only characters left to decode are the characters that we used for formatting.